Automattic/wp-calypso
# Stats: Update paywall upgrade notice

Related to #92724 & pejTkB-1yy-p2.

## Proposed Changes

Updates the error state of the upgrade notice to explicitly call out when the paywall will go into effect if a commercial license is not purchased. Looks like this:

<img width="722" alt="SCR-20240729-phlx" src="https://github.com/user-attachments/assets/860e5001-c1e7-4859-a3c1-579715f4ff99">

Looking for feedback on the final copy before merging.

Current copy:

- title: You need to upgrade to a commercial license to continue using Jetpack Stats
- body: To ensure uninterrupted access to core Stats features, please upgrade to a Jetpack Stats Commercial license using the button below by LOCALIZED-DATE.

Note: The "by LOCALIZED-DATE" is omitted if the API doesn't provide it.

## Why are these changes being made?


If a site has been classified as commercial but does not yet have a commercial license, we grant a grace period before enabling the paywall. This notice is meant to give the user some heads up before they lose access to stats.

## Testing Instructions



* Launch the Calypso Live branch.
* Visit a commercial site without an existing plan.
* Inspect the notice using the React dev tools. Should be an info notice if the blog sticker hasn't been added.
* Toggle the `showPaywallNotice` value in the inspector.
* Confirm you get the "error" version of the notice with more forceful messaging.

## Pre-merge Checklist
